air/src/execution_step/execution_context/scalar_variables.rs

```diff
@@ -118,28 +125,29 @@ impl<'i> Scalars<'i> {
     pub(crate) fn set_value(&mut self, name: impl Into<String>, value: ValueAggregate) -> ExecutionResult<bool> {
         use std::collections::hash_map::Entry::{Occupied, Vacant};
 
-        let shadowing_allowed = self.shadowing_allowed();
-        match self.non_iterable_variables.entry(name.into()) {
+        let name = name.into();
+        let shadowing_allowed = self.shadowing_allowed(&name);
```

Comment:
The problem here is in borrow checker, because `entry` at 130th line takes self by mut and inside its scope it'd be impossible to take another reference to `self`.